src/fountain.rs

```diff
@@ -688,118 +680,135 @@ mod tests {
         // 0x18 is the first byte value that doesn't directly encode a u8,
         // but implies a following value
         assert_eq!(
-            Part::from_cbor(vec![0x18]).unwrap_err().to_string(),
+            Part::from_cbor(vec![0x18].as_slice())
+                .unwrap_err()
+                .to_string(),
             "invalid cbor serialization for Part"
         );
-        // a single cbor item is expected
-        assert_eq!(
-            Part::from_cbor(vec![0x1, 0x1]).unwrap_err().to_string(),
-            "invalid cbor length for Part"
-        );
         // the top-level item must be an array
         assert_eq!(
-            Part::from_cbor(vec![0x1]).unwrap_err().to_string(),
+            Part::from_cbor(vec![0x1].as_slice())
+                .unwrap_err()
+                .to_string(),
             "invalid top-level item"
         );
         // the array must be of length five
         assert_eq!(
-            Part::from_cbor(vec![0x84, 0x1, 0x2, 0x3, 0x4])
+            Part::from_cbor(vec![0x84, 0x1, 0x2, 0x3, 0x4].as_slice())
```

Comment:
`vec![a].as_slice()` could be simplified to `&[a]`. This applies in lots of places.